### PR TITLE
[4.0] fix TinyMCE plugin configuration

### PR DESCRIPTION
**Pull request by Fedik**
Fix for TinyMCE plugin configuration.
In some reason I cannot find `tinymce-builder.js` in 4.0-dev branch, so I copied the file from `staging` manualy.

### Testing Instructions
Check TinyMCE plugin configuration

### Expected result
the styling is broken, you cannot edit the toolbar

### Actual result
the styling looks good, you can edit the toolbar

### CODE REVIEW

**Comment by dgt41** [comment]
I had a PR that for this: https://github.com/joomla/joomla-cms/pull/15371 but I guess it's too big

**Comment by Fedik** [comment]
yes, looks huge 😉

**Comment by dgt41** [comment]
it's ok, as long as it's fixed :)
Couple things here:
You are using sortable and jQuery-ui, but we try to go away from that in core so we use dragulla instead. Do you feel like dropping jquery and jQuery-ui for the builder?
For chosen we are planing to use choices but we're waiting for a feature to be merged...

**Comment by Fedik** [comment]
>  Do you feel like dropping jquery and jQuery-ui for the builder?

Currently I just tried to make it work again 😉 

The thing, that  jQuery-ui is a core of whole builder, so it require a lot rewriting.
Maybe sometime later.
Also will be need some time to check how the dragula.js works with multiple instances.

> For chosen we are planing 

I am fine with native `<select>` 😄

**Comment by dgt41** [comment]
> I am fine with native <select>

So do I, but there are few places (tags, modules, categories on the fly) that we need a better ui or some search functionality, so there is a legitimate case using javascript driven element there.
The best solution will be to find a custom element for that but I haven't see anything that fits our use case so far

File: media/editors/tinymce/js/tinymce-builder.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+
+    /**
+     * Create the element needed for renderBar()
+     * @param {String} name
+     * @param {Object} info
+     * @param {String} type
+     *
+     * @return {jQuery}
+     *
+     * @since  3.7.0
```
Comment (zero-24) [inline media/editors/tinymce/js/tinymce-builder.js]: 3.7.0 or 4.0?

File: media/editors/tinymce/js/tinymce-builder.js [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+    /**
+     * Render the toolbar/menubar
+     *
+     * @param {HTMLElement} container  The toolbar container
+     * @param {String}      type       The type toolbar or menu
+     * @param {Array|null}  value      The value
+     * @param {Boolean}     withInput  Whether append input
+     *
+     * @since  3.7.0
```
Comment (zero-24) [inline media/editors/tinymce/js/tinymce-builder.js]: 3.7.0 or 4.0?

File: media/editors/tinymce/js/tinymce-builder.js [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+    };
+
+    /**
+     * Joomla TinyMCE Builder
+     *
+     * @param {HTMLElement} container
+     * @param {Object}      options
+     * @constructor
+     *
+     * @since  3.7.0
```
Comment (zero-24) [inline media/editors/tinymce/js/tinymce-builder.js]: 3.7.0 or 4.0?

File: media/editors/tinymce/js/tinymce-builder.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+        return $element;
+    };
+
+    /**
+     * Append input to the button item
+     * @param {HTMLElement} element
+     * @param {String}      group
+     * @param {String}      set
+     *
+     * @since  3.7.0
```
Comment (zero-24) [inline media/editors/tinymce/js/tinymce-builder.js]: 3.7.0 or 4.0?

**Review by zero-24 — COMMENTED** [review]
3.7.0 or 4.0? As your PR is against 4.0 ;) Maybe __DEPLOY_VERSION__ but we should first check if that script is running against js files-

File: media/editors/tinymce/js/tinymce-builder.js (hunk 3 above)
Comment (dgt41) [inline media/editors/tinymce/js/tinymce-builder.js]: This is not a new file, it's forgotten in one of the syncs. 3.7 is the correct one

**Pull request merged**